src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs

```diff
@@ -20,22 +20,39 @@ use bytes::Bytes;
 use crate::hummock::iterator::variants::*;
 use crate::hummock::iterator::HummockIterator;
 use crate::hummock::value::HummockValue;
-use crate::hummock::{key, HummockResult};
+use crate::hummock::{key, HummockEpoch, HummockResult};
+use crate::storage_value::VALUE_META_SIZE;
 
 pub(super) type SharedBufferItem = (Bytes, HummockValue<Bytes>);
 
 /// A write batch stored in the shared buffer.
 #[derive(Clone, Debug)]
 pub struct SharedBufferBatch {
     pub(super) inner: Arc<[SharedBufferItem]>,
-    pub(super) epoch: u64,
+    pub(super) epoch: HummockEpoch,
+    pub(super) size: u64,
 }
 
 impl SharedBufferBatch {
-    pub fn new(sorted_items: Vec<SharedBufferItem>, epoch: u64) -> Self {
+    pub fn new(sorted_items: Vec<SharedBufferItem>, epoch: HummockEpoch) -> Self {
+        // size = Sum(length of full key + length of user value)
+        let size: u64 = sorted_items
+            .iter()
+            .map(|(k, v)| {
+                let vsize = {
+                    match v {
+                        HummockValue::Put(_, val) => VALUE_META_SIZE + val.len(),
+                        HummockValue::Delete(_) => VALUE_META_SIZE,
```

Comment:
What if the inner implementation of `HummockValue` has changed? IIRC, there is a function of `encoded_len`.

Reply:
`encoded_len` seems to be the length in the SST file. But we need the size of kv pair in memory.